var/spack/repos/builtin/packages/py-kosh/package.py

```diff
@@ -28,3 +30,13 @@ class PyKosh(PythonPackage):
     depends_on("py-llnl-sina@1.11:", type=("build", "run"))
     depends_on("py-networkx", type=("build", "run"))
     depends_on("py-numpy", type=("build", "run"))
+
+    with when("@3:"):
+        depends_on("py-matplotlib")
+        depends_on("py-scipy")
+        depends_on("py-h5py@3:")
+        depends_on("py-scikit-learn@1.0.2:")
+        depends_on("py-pandas")
+        depends_on("py-hdbscan@0.8.29:")
```

Comment:
Where does 0.8.29+ come from?

Reply:
As I understand it from @doutriaux1 , previous revisions failed regression tests in py-kosh. 0.8.29 is the oldest that's been tested and confirmed working.

Reply:
yes we need 0.8.29